language/move-core/types/src/account_address.rs

```diff
@@ -70,6 +70,19 @@ impl AccountAddress {
 
         AccountAddress::try_from(padded_result)
     }
+
+    // Note: This is inconsistent with current types because AccountAddress is derived
+    // from consensus key which is of type Ed25519PublicKey. Since AccountAddress does
+    // not mean anything in a setting without remote authentication, we use the network
+    // public key to generate a peer_id for the peer.
+    // See this issue for potential improvements: https://github.com/libra/libra/issues/3960
+    pub fn from_identity_public_key(identity_public_key: x25519::PublicKey) -> Self {
```

Comment:
this should be a `From` impl

```rust
impl From<x25519::PublicKey> for AccountAddress {
    fn from(pubkey: x25519::PublicKey) -> Self {
        let mut array = [0u8; Self::LENGTH];
        let pubkey_slice = identity_public_key.as_slice();
        // keep only the last 16 bytes
        array.copy_from_slice(&pubkey_slice[x25519::PUBLIC_KEY_SIZE - Self::LENGTH..]);
        Self(array)
    }
}
```

then we can do `PeerId::from(remote_static_pubkey)`

Reply:
intuitively, it feels cleaner if this is a custom function as it indicates that it is not doing something straight forward (a hack) 

(as opposed to `From` which usually sound to me like they're just convenience functions)

what do you think?